src/highrl/envs/env_generator.py

```diff
@@ -0,0 +1,248 @@
+"""Implementation for the environment to train the environmenet generator modle
+
+The model is assumed to take the difficulty (generated by the teacher policy) and
+generate a robot-environment abiding by the generated difficulty.
+
+In this module, we implement the environment generator using reinforcement learninig,
+for not constraining the creativity the model, i.e. there are numerous ways to generate
+environments that abide by a certain difficulty.
+"""
+from typing import Tuple, Callable, List
+import logging
+import random
+from gym import Env, spaces
+import numpy as np
+import torch
+from torch.utils.tensorboard import SummaryWriter  # type: ignore
+from torch import nn
+from stable_baselines3 import PPO  # type: ignore
+from stable_baselines3.common.policies import ActorCriticPolicy
+
+from highrl.obstacle.obstacles import Obstacles
+from highrl.obstacle.single_obstacle import SingleObstacle
+from highrl.agents.robot import Robot
+from highrl.utils import Position
+from highrl.utils.teacher_checker import compute_difficulty
+from highrl.utils.logger import init_logger
+
+_LOG = logging.getLogger(__name__)
+
+
+class EnvGeneratorPolicy(nn.Module):
+    """Implementation for the environment generator model"""
+
+    hidden_size = 16
+    hard_obs = 2
+    med_obs = 3
+    small_obs = 4
+    device = "cuda"
+    output_size = 4
+
+    def __init__(
+        self,
+        feature_dim: int = 1,
+        last_layer_dim_pi: int = 40,
```

Comment:
Did the `last_layer_dim_pi` came from `(hard_obs  + med_obs + small_obs ) * output_size ` ?

Reply:
`(hard_obs + med_obs  + small_obs + 1) * output_size`.

Added a comment in the code for that note.

Reply:
Great.

Reply:
Why the +1 ?